hydroflow_datalog/src/snapshots/hydroflow_datalog__tests__join_with_other.snap

```diff
@@ -35,22 +38,33 @@ fn main() {
                             std::task::Poll::Ready(maybe) => maybe,
                             std::task::Poll::Pending => None,
                         });
+                    hydroflow::assert_var_impl!(op_8v1: std::iter::Iterator);
                     let op_3v1 = op_8v1;
+                    hydroflow::assert_var_impl!(op_3v1: std::iter::Iterator);
                     let op_4v1 = op_3v1;
+                    hydroflow::assert_var_impl!(op_4v1: std::iter::Iterator);
                     let op_12v1 = op_2v1.map(|v: (_, _)| ((v.0, v.1), v));
+                    hydroflow::assert_var_impl!(op_12v1: std::iter::Iterator);
                     let op_13v1 = op_4v1.map(|v: (_, _)| ((v.1, v.0), v));
+                    hydroflow::assert_var_impl!(op_13v1: std::iter::Iterator);
```

Comment:
I wonder, does this help with type inference in Rust Analyzer? Noticed that it's only able to typecheck some expansions.

Reply:
Shouldn't change type inference, but will help with error message locality in some situations. Some iterators/pusherators have "late" trait bounds, i.e. trait bounds that appear on `impl Iterator for X` but not on `struct X`. So this makes sure the error messages appear as soon as possible in the right places